### ISSUE TITLE
Specifying static efficiencies for runs

### ISSUE THREAD

**Issue by Uldreth** [dkouzoup/ms-eetc#8]
I am not sure if this is a bug or just something that is _really_ not obvious, but I have encountered some issues in modifying the simulation for figure5.

In what follows, I only talk about computations with _static efficiency_.

When running figure5.py, it appears that the static efficiencies \eta^{tr} and \eta^{rb} both concide and are specified by the variable `etaMax` on line 94 with value 0.73. I wanted to change it to the pair of efficiencies \eta^{tr} = 0.875 and \eta^{rb} = 0.7 found in the train model file NL_Intercity_VIRM6.json.

So what I did is that in line 97 (definition of `fun1`), I have changed the values of `etaMax` to two literals, e.g. `fun1 = lambda f,v: f*v*(f>0)*(1 - 0.875)/0.875 - (1-0.7)*f*v*(f<0)`, where essentially \eta^{tr} = 0.875 and \eta^{rb} = 0.7. Strangely enough, on the generated plots, the consumption values were exactly the same in almost all runs. In the following I provide an enumeration of these results:

- \eta^{tr} = \eta^{rb} = etaMax = 0.73:  151.1 / 124.6 / 111.8 / 106.4 kWh (figure5_eta_max.pdf)
- \eta^{tr} = 0.875, \eta^{rb} = 0.7: 151.1 / 124.6 / 111.8 / 106.4 kWh (figure5_mixed_eff_875_700.pdf) 
- \eta^{tr} = 0.001, \eta^{rb} = 0.001:  151.1 / 124.6 / 111.9 / 106.5 kWh (figure5_almost_zero.pdf)
- \eta^{tr} = 1, \eta^{rb} = 1: 151.1 / 125.9 / 112.5 / 106.4 kWh (figure5_perfect.pdf)

These show that the consumptions are barely sensitive to the change in efficiency. However if I comment out line 134, thus forcing the program to use the line `powerLosses = lambda f,v: f*v*(f>0)*(1 - self.etaTraction)/self.etaTraction - (1-self.etaRgBrake)*f*v*(f<0)` (line 203 in train.py) to define the power losses (i.e. the \eta^{tr} and \eta^{rb} from the train model file), then the consumptions become

- 151.1 / 120.6 / 107.6 / 100.9 kWh (figure5_bypass.pdf), which now completely coincide with those of the dynamic efficiency. So this makes me think that in this case we run with dynamic efficiency.

There is either a bug here or I severely misunderstand how the program processes static efficiencies. How can I run figure5.py while configuring arbitrary etaTraction and eta RgBrake?

**Comment by Uldreth** [comment]
[figure5_almost_zero.pdf](https://github.com/dkouzoup/ms-eetc/files/15130292/figure5_almost_zero.pdf)
[figure5_bypass.pdf](https://github.com/dkouzoup/ms-eetc/files/15130293/figure5_bypass.pdf)
[figure5_eta_max.pdf](https://github.com/dkouzoup/ms-eetc/files/15130294/figure5_eta_max.pdf)
[figure5_mixed_eff_875_700.pdf](https://github.com/dkouzoup/ms-eetc/files/15130296/figure5_mixed_eff_875_700.pdf)
[figure5_perfect.pdf](https://github.com/dkouzoup/ms-eetc/files/15130298/figure5_perfect.pdf)

**Comment by dkouzoup** [comment]
Hello Uldreth, thank you for the input and apologies for the confusion. The way you intended to change the code is in principle correct. There might be some confusion with the interpretation of the legends. The two numbers in the parentheses refer to _real_ energy consumption and _real_ energy losses, where with _real_ we mean using the dynamic losses function (fun2). So it could be that because fig5 is a short simulation of an acceleration phase, changing the static efficiency does not really affect the optimal trajectory, and if the trajectory remains the same, the numbers in the parenthesis will stay the same too. Did you try to change the static efficiency factor in one of the other plots? I believe that for a complete journey changing the static efficiency will in fact have a noticeable impact on the the optimal trajectory. 

Does this make sense?

**Comment by Uldreth** [comment]
Hello, thank you for getting back to me! Yes, it makes sense and it answers my question/problem. For some reason I thought that the consumptions/losses would be computed with the same loss function used in the OCP. Obviously, it makes sense this way as it allows for comparisons of the degree of inefficiency introduced by the simplifications of the model.

I wanted to understand the losses computed from the static efficiency model. Once you cleared up this misunderstanding on my part, all I had to do was to change fun2 to fun1 in the relevant part of the code!

**Comment by dkouzoup** [comment]
Great, glad we sorted it out!
